src/platform/tests/BUILD.gn

```diff
@@ -89,6 +89,11 @@ if (chip_device_platform != "none" && chip_device_platform != "fake") {
     if (chip_device_platform == "linux") {
       test_sources += [ "TestConnectivityMgr.cpp" ]
     }
+
+    if (chip_enable_ota_requestor) {
+      test_sources += [ "TestGenericOTARequestorStorage.cpp" ]
+      public_deps += [ "${chip_root}/src/lib/asn1" ]
```

Comment:
Why asn.1 ?

Reply:
There's a whole dependency chain from cluster headers and auto-generated headers generated by asn1. The code doesn't compile without that.

Reply:
That dependency is just injecting the unfortunate transitive dependency that actually brings in the world and almost makes platform circular to data model. This is why above I am saying to avoid ANY generated code dependencies directly inside the platform/device layer. We are one or two files away from becoming circular.

Reply:
It turns out this was actually by some unit test helper which I cleaned up and this dependency is no longer needed.